### PR TITLE
run tests using actions

### PR DESCRIPTION
**Pull request by kattrali**
## Goal

1. Make tests work
2. Make tests run faster

## Changeset

* Updated tests to be root-directory-agnostic. Some of the assertions depended on the tests running from `$GOPATH/.../`

### CODE REVIEW

File: .github/workflows/test-package.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+on: [ push, pull_request ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: ['1.9', '1.10', '1.11', '1.12', '1.13', '1.14', '1.15']
```
Comment (imjoehaines) [inline .github/workflows/test-package.yml:11]: Do we need to go back as far as 1.7 like Travis did?

File: bugsnag_test.go [hunk 2]
```diff
@@ -612,15 +612,14 @@ func assertValidSession(t *testing.T, event *simplejson.Json, unhandled bool) {
 }
 
 func verifyExistsInStackTrace(t *testing.T, exception *simplejson.Json, exp *stackFrame) {
-	isFile := func(frame *simplejson.Json) bool { return getString(frame, "file") == exp.File }
+	isFile := func(frame *simplejson.Json) bool { return strings.HasSuffix(getString(frame, "file"), exp.File) }
 	isMethod := func(frame *simplejson.Json) bool { return getString(frame, "method") == exp.Method }
 	isLineNumber := func(frame *simplejson.Json) bool { return getInt(frame, "lineNumber") == exp.LineNumber }
-	isInProject := func(frame *simplejson.Json) bool { return getBool(frame, "inProject") == exp.InProject }
```
Comment (imjoehaines) [inline bugsnag_test.go:618]: Doesn't this change mean we're not testing if a frame is in project anymore?

File: configuration_test.go [hunk 3]
```diff
@@ -97,16 +97,18 @@ func TestIsProjectPackage(t *testing.T) {
...
 	gopath := os.Getenv("GOPATH")
+	Configure(Configuration{
+		ProjectPackages: []string{
+			"main",
+			"star*",
+			"example.com/a",
+			"example.com/b/*",
+			"example.com/c/**",
+		},
+        SourceRoot: gopath + "/src/",
```
Comment (imjoehaines) [inline configuration_test.go:109]: Mixed spaces & tabs, maybe it's worth adding `go fmt` to CI too (or just running it locally :d)?

**Review by imjoehaines — CHANGES_REQUESTED** [review]

File: bugsnag_test.go (hunk 2 above)
Comment (kattrali) [reply bugsnag_test.go:618]: Yes and no. These test assertions were dependent on the location of the package while being tested, which made them fail unexpectedly if not in `$GOPATH/src`. There are several test cases for `Configuration.isProjectPackage()`, which underpins the `InProject` behavior. I'm also adding more tests as a part of some internals changes to resolve issues with `CallersFrames`.

Comment (imjoehaines) [reply bugsnag_test.go:618]: 👍 that makes sense

**Review by imjoehaines — APPROVED** [review]

**Pull request merged**